### DISCUSSION TITLE
Setting tailwind css styles using string

### DISCUSSION THREAD

**Discussion by eugener** [zed-industries/zed#8644] [Feature Requests]
Would it be of any benefit to set `gpui` component styles as a `&str`, which in turn would call exiting style APIs?
This, in theory, would enable an ability to create style independent components and even minimize the amount of code needed to define component rendering.
It should be even possible to do the TailwindCSS style check at compile time. 

Here is what it might look like:

```
  div().style("flex justify-center text-3xl")
```
which would convert to

```
  div()
   .flex()
   .justify_center()
   .text_3xl()
```

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
